api/envoy/extensions/filters/network/meta_protocol_proxy/matcher/action/v3/action.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.meta_protocol_proxy.matcher.action.v3;
+
+import "envoy/config/route/v3/route_components.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.meta_protocol_proxy.matcher.action.v3";
+option java_outer_classname = "RouteActionProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: Meta Protocol Proxy Route Action Configuration]
+
+// Configuration for the route match action.
+// [#not-implemented-hide:]
+message RouteAction {
+  oneof cluster_specifier {
+    option (validate.required) = true;
+
+    // Indicates the upstream cluster to which the request should be routed.
+    string cluster = 1;
+
+    // Multiple upstream clusters can be specified for a given route. The request is routed to one
+    // of the upstream clusters based on weights assigned to each cluster.
+    // Currently ClusterWeight only supports the name and weight fields.
+    config.route.v3.WeightedCluster weighted_clusters = 2;
+  }
+  //TODO other configuration: RetryPolicy, Mirroring, Fault Injection, etc.
+}
```

Comment:
What's the long term plan here? How much of the existing `RouteAction` in RDS will be reproduced? If it's the entire thing, then arguably we should just augment existing `RouteAction` with the ability to match on MetaProtocol attributes (maybe it can already via dynamic metadata matchers?). If not, we should be clear on the long term relationship across these structures.

Reply:
We will check it.

Reply:
@htuch @wbpcode 

I had the same thought about that: should we reuse the existing RouteAction? Or create another similar one? 

My concern is the existing RouteAction is highly coupled with HTTP protocols. If we reuse it, some properties or inner structures may become ambiguous for MetaProtocol. 

```
{
  "cluster": "...",
  "cluster_header": "...",
  "weighted_clusters": "{...}",
  "cluster_not_found_response_code": "...",
  "metadata_match": "{...}",
  "prefix_rewrite": "...",
  "regex_rewrite": "{...}",
  "host_rewrite_literal": "...",
  "auto_host_rewrite": "{...}",
  "host_rewrite_header": "...",
  "host_rewrite_path_regex": "{...}",
  "timeout": "{...}",
  "idle_timeout": "{...}",
  "retry_policy": "{...}",
  "request_mirror_policies": [],
  "priority": "...",
  "rate_limits": [],
  "include_vh_rate_limits": "{...}",
  "hash_policy": [],
  "cors": "{...}",
  "max_grpc_timeout": "{...}",
  "grpc_timeout_offset": "{...}",
  "upgrade_configs": [],
  "internal_redirect_policy": "{...}",
  "internal_redirect_action": "...",
  "max_internal_redirects": "{...}",
  "hedge_policy": "{...}",
  "max_stream_duration": "{...}"
}
```

Reply:
@zhaohuabing I think so too. May be we should create new RouteAction which is protocol-independent for the meta protocol proxy.  cc @htuch

Reply:
I agree it's not great to have the duplication, but I would be OK starting from scratch and making sure we don't have ambiguity.

Reply:
https://github.com/cncf/xds/blob/cb28da3451f158a947dfc45090fe92b07b243bc1/xds/type/matcher/v3/matcher.proto#L38-L39

I think different protocols differ on `RouteAction`, would it be better if we can provide a public interface similar to `action` in the generic matching?

Reply:
If it's necessary, we can add new `custom_protocol_specific_config` field to the `RouteAction` in the future.